slyd/slyd/splash/ferry.py

```diff
@@ -136,10 +136,13 @@ def log(self, s):
     @pyqtSlot('QString')
     def sendMessage(self, message):
         message = text(message)
+        message = message.strip('\x00') # Allocation bug somewhere leaves null characters at the end.
         try:
             command, data = json.loads(message)
-        except ValueError:  # XXX: Possibly null terminated string
-            command, data = json.loads(message[:-1])
+        except ValueError as e:
+            return log.err(ValueError(
```

Comment:
This is called from javascript, so the traceback is just a couple of pyqt
frames and nothing from our app, not very useful.
On Oct 1, 2015 2:30 PM, "Michał Modzelewski" <notifications@github.com>
wrote:

> In slyd/slyd/splash/ferry.py
> <https://github.com/scrapinghub/portia/pull/292#discussion_r40885600>:
>
> >          try:
> >              command, data = json.loads(message)
> > -        except ValueError:  # XXX: Possibly null terminated string
> > -            command, data = json.loads(message[:-1])
> > +        except ValueError as e:
> > +            return log.err(ValueError(
>
> id prefer the traceback to go to sentry, it would allow to see where this
> is coming from.
> just log.err() here will create a twisted failure with the current
> exc_info()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scrapinghub/portia/pull/292/files#r40885600>.
>